src/package.h

```diff
@@ -43,6 +46,33 @@ class Package {
     std::string strategy_;
 };
 
+/// Returns optimal fill mass for a resource to be packaged. Can be used
+/// to determine how to respond to requests for material, and to actually
+/// package and send off trades
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg);
+
+/// Repackages a single resource into a package. If some quantity of the 
+/// resource cannot be packaged using the given packaging strategy and
+/// restrictions, the remainder is left in the resource object. 
+template <class T>
+std::vector<typename T::Ptr> Repackage(typename T::Ptr r, Package::Ptr pkg) {
```

Comment:
Is there any reason you have this in the .h file instead of the .cc?

Reply:
Not really